flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityConfiguration.java

```diff
@@ -42,10 +44,18 @@
  */
 public class SecurityConfiguration {
 
-	private static final List<SecurityModuleFactory> DEFAULT_MODULES = Collections.unmodifiableList(
-		Arrays.asList(new HadoopModuleFactory(), new JaasModuleFactory(), new ZookeeperModuleFactory()));
+	private static final String DEFAULT_CONTEXT_FACTORY = DefaultSecurityContextFactory.class.getCanonicalName();
 
-	private final List<SecurityModuleFactory> securityModuleFactories;
+	private static final List<String> DEFAULT_MODULE_FACTORIES = Collections.unmodifiableList(Arrays.asList(
+		HadoopModuleFactory.class.getCanonicalName(),
+		JaasModuleFactory.class.getCanonicalName(),
+		ZookeeperModuleFactory.class.getCanonicalName()));
+
+	private final String securityContextFactory;
+
+	private final List<String> securityModuleFactories;
+
+	private final Map<String, Object> properties;
```

Comment:
Why are the properties needed? Is it only to make the test work?

Reply:
hmm. you are absolutely right -> do you suggest I just use flinkConfig for storing/retrieving the additional properties?

Reply:
if that works, without changing the production code for the test, then yes.